pkg/operator/handler.go

```diff
@@ -295,8 +311,10 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 		}
 
 		if cr.Spec.Replicas == o.Status.ReadyReplicas {
+			h.operatorStatus(osapi.OperatorAvailable, osapi.ConditionTrue, "")
```

Comment:
shouldn't we report available as long as there's at least 1 replica up?

Reply:
I think we shouldn't, since our resource asked for more relicas.

Reply:
It's available.  This would be a case where setting "Progressing=true" might make sense.

Available=true  (the registry is available, you can access it).  Available does not mean "matches your desired state", only that it's accessible.

Progressing=true (we are still in the process of applying your desired configuration to the registry).

Progressing should also be set to true when there is a deployment in progress for any reason.  When the deployment is complete, Progressing would be set to false.

Reply:
@legionus tagging you in this thread just to make sure it's not missed since it's a comment on an obsolete commit.